### ISSUE TITLE
Hive "es.read.field.as.array.include" setting for fields that are sometimes array or string

### ISSUE THREAD

**Issue by MainAero** [elastic/elasticsearch-hadoop#945]
Using Hadoop 2.7.1, Hive 2.1.1, ES-Hadoop 5.2.1

Like in #856 I've fields in my dataset that are sometimes array (array of strings) or string.

Example:
```bash
curl -XGET "http://localhost/index/_search"
{
...
"_source":{"FIELDA": "value"}
...
"_source":{"FIELDA": ["value", "value2"]}
}
```

In hive I created an external table:
```sql
CREATE EXTERNAL TABLE mytable (
     fielda ARRAY<STRING>
)
STORED BY 'org.elasticsearch.hadoop.hive.EsStorageHandler'
TBLPROPERTIES(
              'es.resource' = 'index',
              'es.nodes' = 'es',
              'es.port' = '9200',
              'es.mapping.names' = 'fielda:FIELDA',
              'es.read.field.as.array.include' = 'FIELDA')
```

As you can see I'm using the `es.read.field.as.array.include` to expect that the field `FIELDA` will be always read as array.

But when I execute a query: 
```sql 
SELCT fielda FROM mytable;
```
it throws this exception:
```java
org.apache.hive.service.cli.HiveSQLException: java.io.IOException: java.lang.ClassCastException: org.apache.hadoop.io.Text cannot be cast to org.apache.hadoop.io.ArrayWritable
```

**Comment by jbaiera** [comment]
Definitely looks like the same issue as #856. I've updated the tags on the ticket and will link back here when a fix is available. Thanks for reporting @MainAero!

**Comment by jbaiera** [comment]
#856 has been merged in and the fix for this should be available as of v5.6.5. Closing
